src/test/java/core/basesyntax/service/impl/FruitShopServiceImplTest.java

```diff
@@ -0,0 +1,82 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.FruitShopService;
+import core.basesyntax.strategy.handler.OperationHandler;
+import core.basesyntax.strategy.handler.impl.BalanceHandler;
+import core.basesyntax.strategy.handler.impl.OperationStrategyImpl;
+import core.basesyntax.strategy.handler.impl.PurchaseHandler;
+import core.basesyntax.strategy.handler.impl.ReturnHandler;
+import core.basesyntax.strategy.handler.impl.SupplyHandler;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class FruitShopServiceImplTest {
+    private static final String APPLE = "apple";
+    private static final String BANANA = "banana";
+    private static final String ORANGE = "orange";
+    private static final String PEAR = "pear";
+    private FruitShopService fruitShopService;
+
+    @BeforeEach
+      public void setUp() {
+        Map<FruitTransaction.Operation, OperationHandler> operationHandlerMap
+                = new HashMap<>();
+        operationHandlerMap.put(FruitTransaction.Operation.BALANCE, new BalanceHandler());
+        operationHandlerMap.put(FruitTransaction.Operation.PURCHASE, new PurchaseHandler());
+        operationHandlerMap.put(FruitTransaction.Operation.RETURN, new ReturnHandler());
+        operationHandlerMap.put(FruitTransaction.Operation.SUPPLY, new SupplyHandler());
+        fruitShopService = new
+              FruitShopServiceImpl(new OperationStrategyImpl(operationHandlerMap));
+        Storage.storage.clear();
+        Storage.storage.put(APPLE, 10);
+        Storage.storage.put(BANANA, 20);
+        Storage.storage.put(ORANGE, 30);
+    }
+
+    @AfterEach
+    public void tearDown() {
+        Storage.storage.clear();
+    }
+
+    @Test
+    public void test_ProcessData_UpdateStorage_Ok() {
```

Comment:
In your test methods please separate your code for better readability. There is a common convention to have empty line between setup logic, action logic and assertion logic.